Factor out hasSingleLine logic into a method

fantasai opened the pull request:
#### e2cc160e3d0d51b1fb74d0786abf400a181946ce
<pre>
Factor out hasSingleLine logic into a method
<a href="https://bugs.webkit.org/show_bug.cgi?id=286311">https://bugs.webkit.org/show_bug.cgi?id=286311</a>
<a href="https://rdar.apple.com/143332919">rdar://143332919</a>

Reviewed by Alan Baradlay.

Moves hasSingleLine logic into a helper method on InlineIterator::InlineBox.

* Source/WebCore/layout/integration/inline/InlineIteratorInlineBox.h:
(WebCore::InlineIterator::InlineBox::isSplit const):
* Source/WebCore/rendering/InlineBoxPainter.cpp:
(WebCore::InlineBoxPainter::paintMask):
(WebCore::InlineBoxPainter::paintDecorations):
(WebCore::InlineBoxPainter::paintFillLayer):
(WebCore::InlineBoxPainter::paintBoxShadow):

Canonical link: <a href="https://commits.webkit.org/289392@main">https://commits.webkit.org/289392@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/29cded33c60873b8d0895e117859530d813b4f3b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/86782 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/6289 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/41121 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/91629 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/37514 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/88831 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/6557 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/14349 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/67085 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/24854 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/89785 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/4991 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/78553 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/47406 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/4776 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/32910 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/36632 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/75284 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/33795 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/93523 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/13935 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/10105 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/75888 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/14136 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/74399 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/75083 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18468 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/19398 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/17810 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/6720 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/13958 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/19218 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/13696 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/17141 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/15481 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

fantasai commented on Source/WebCore/layout/integration/inline/InlineIteratorInlineBox.h:
```diff
@@ -83,6 +84,11 @@ inline InlineBoxIterator InlineBox::iterator() const
+{
+    return !nextInlineBoxLineRightward() && !nextInlineBoxLineLeftward();
+}
+
```
Wasn't sure what to name it, this seemed reasonable, but open to other ideas.

alanbaradlay replied:
non-spanning is what we have in IFC but I should not be making comments about naming things.

fantasai replied:
Gecko would probably call it something like `hasContinuation`...

alanbaradlay replied:
that would clash with our definition of continuation (which should hopefully go away sometime soon)

alanbaradlay replied:
though I still prefer it over "singular"

fantasai replied:
Oh, what about `isSplit()` (returning the inverse)?

alanbaradlay replied:
that sounds reasonable

alanbaradlay commented on Source/WebCore/layout/integration/inline/InlineIteratorInlineBox.h:
```diff
@@ -51,6 +51,7 @@ class InlineBox : public Box {
     LeafBoxIterator firstLeafBox() const;
     LeafBoxIterator lastLeafBox() const;
     LeafBoxIterator endLeafBox() const;
+    inline bool isSingular() const;
```
? :)

fantasai replied:
🤦‍♀️

fantasai committed:
Factor out hasSingleLine logic into a method
https://bugs.webkit.org/show_bug.cgi?id=286311
rdar://143332919

Reviewed by Alan Baradlay.

Moves hasSingleLine logic into a helper method on InlineIterator::InlineBox.

* Source/WebCore/layout/integration/inline/InlineIteratorInlineBox.h:
(WebCore::InlineIterator::InlineBox::isSplit const):
* Source/WebCore/rendering/InlineBoxPainter.cpp:
(WebCore::InlineBoxPainter::paintMask):
(WebCore::InlineBoxPainter::paintDecorations):
(WebCore::InlineBoxPainter::paintFillLayer):
(WebCore::InlineBoxPainter::paintBoxShadow):

Canonical link: https://commits.webkit.org/289392@main
